Passing a const in optional::value_or() not compiling

The following code fails to compile, while std::optional compiles fine:
```c++
const int DEFAULT = 0;
int res = etl::optional<int>{}.value_or(DEFAULT);
```

```sh
etl/optional.h:1099:66: error: binding reference of type ‘std::remove_reference<int>::type&’ {aka ‘int&’} to ‘const int’ discards qualifiers
 1099 |         return has_value() ? etl::move(value()) : etl::forward<T>(default_value);
      |                                                   ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
```

I believe it's because in the implementation of `value_or`, the template parameter T of the optional class is forwarded instead of the template parameter U of the method.

Here is a compiler explorer link to illustrate: https://godbolt.org/z/TavvKzhnr

--- a/include/etl/optional.h
+++ b/include/etl/optional.h
@@ -470,7 +470,7 @@ namespace etl
       etl::enable_if_t<etl::is_convertible<U, T>::value, T>
         value_or(U&& default_value) const&
       {
-        return has_value() ? value() : etl::forward<T>(default_value);
+        return has_value() ? value() : static_cast<T>(etl::forward<U>(default_value));
       }
 
       //***************************************************************************
@@ -481,7 +481,7 @@ namespace etl
       etl::enable_if_t<etl::is_convertible<U, T>::value, T>
         value_or(U&& default_value)&&
       {
-        return has_value() ? etl::move(value()) : etl::forward<T>(default_value);
+        return has_value() ? etl::move(value()) : static_cast<T>(etl::forward<U>(default_value));
       }
 #endif
 
@@ -1084,7 +1084,7 @@ namespace etl
       etl::enable_if_t<etl::is_convertible<U, T>::value, T>
         value_or(U&& default_value) const&
       {
-        return has_value() ? value() : etl::forward<T>(default_value);
+        return has_value() ? value() : static_cast<T>(etl::forward<U>(default_value));
       }
 
       //***************************************************************************
@@ -1095,7 +1095,7 @@ namespace etl
       etl::enable_if_t<etl::is_convertible<U, T>::value, T>
         value_or(U&& default_value)&&
       {
-        return has_value() ? etl::move(value()) : etl::forward<T>(default_value);
+        return has_value() ? etl::move(value()) : static_cast<T>(etl::forward<U>(default_value));
       }
 #endif
 

--- a/test/test_optional.cpp
+++ b/test/test_optional.cpp
@@ -299,6 +299,21 @@ namespace
     }
 
     //*************************************************************************
+    TEST(test_value_or_const)
+    {
+      using FundamentalType = int;
+      using NonFundamentalType = std::string;
+
+      const FundamentalType constFT{ 5 };
+      int resultFT = etl::optional<FundamentalType>{}.value_or(constFT);
+      CHECK_EQUAL(5, resultFT);
+
+      const NonFundamentalType constNFT{ "Default" };
+      NonFundamentalType resultNFT = etl::optional<NonFundamentalType>{}.value_or(constNFT);
+      CHECK_EQUAL("Default", resultNFT);
+    }
+
+    //*************************************************************************
     struct github_bug_720_bug_helper
     {
       int value{ 5 };

